src/repr/type_binary.mli

```diff
@@ -17,6 +17,27 @@
 open Type_core
 open Staging
 
+module Uuid : sig
+  type t = string
+
+  val t : t ty
+  val of_string : string -> t
+end
+
+module Shape : sig
+  type t
+  (** The type of {i shapes} of a particular type with respect to the binary
+      decoders defined here. Shapes are represented canonically such that
```

Comment:
Why only the binary ones?

Reply:
Because the non-binary ones would have different shapes. e.g. for a Json codec we need to care about the names of fields.